frontend/src/components/filter/goalFilters.js

```diff
@@ -0,0 +1,102 @@
+/* eslint-disable import/prefer-default-export */
+import React from 'react';
+import moment from 'moment';
+import { formatDateRange } from '../../utils';
+import {
+  DATE_CONDITIONS,
+  FILTER_CONDITIONS,
+} from '../../Constants';
+import FilterDateRange from './FilterDateRange';
+import FilterReasonSelect from './FilterReasonSelect';
+import FilterTopicSelect from './FilterTopicSelect';
+import FilterStatus from './FilterStatus';
+
+const YEAR_TO_DATE = formatDateRange({
+  yearToDate: true,
+  forDateTime: true,
+});
+
+const EMPTY_MULTI_SELECT = {
+  Is: [],
+  'Is not': [],
+};
+
+const handleArrayQuery = (q) => {
+  if (q.length) {
+    return [q].flat().join(', ');
+  }
+  return '';
+};
+
+export const createDateFilter = {
+  id: 'createDate',
+  display: 'Create date',
+  conditions: DATE_CONDITIONS,
+  defaultValues: {
+    'Is within': YEAR_TO_DATE,
+    'Is after': '',
+    'Is before': '',
+  },
+  displayQuery: (query) => {
+    if (query.includes('-')) {
+      return formatDateRange({
+        string: query,
+        withSpaces: false,
+      });
+    }
+    return moment(query, 'YYYY/MM/DD').format('MM/DD/YYYY');
+  },
+  renderInput: (id, condition, query, onApplyQuery) => (
+    <FilterDateRange
+      condition={condition}
+      query={query}
+      updateSingleDate={onApplyQuery}
+      onApplyDateRange={onApplyQuery}
+    />
+  ),
+};
+
+export const reasonsFilter = {
+  id: 'reason',
+  display: 'Reasons',
+  conditions: FILTER_CONDITIONS,
+  defaultValues: EMPTY_MULTI_SELECT,
+  displayQuery: handleArrayQuery,
+  renderInput: (id, condition, query, onApplyQuery) => (
+    <FilterReasonSelect
+      inputId={`reason-${condition}-${id}`}
+      onApply={onApplyQuery}
+      query={query}
+    />
+  ),
+};
+
```

Comment:
@jasalisbury Building on your suggestions, I broke apart the filters and they are now imported into individual files. All the actual display components still live in the root of the /filter folder. Right now, there is only need for one version of each as far as React components, if that makes sense, so it feels a bit premature to sort them. (They'd largely go in one "shared" folder right now, I think.)

I figured that in these ```activityReportFilters```, ```goalFilters```, etc, where objects are currently being exported we could convert them to functions as necessary, and that would allow us to do the configuration in the constant files where they are being imported on a per page basis. In the example of the status filter, for example, we could create a function that passes in the options for the multiselect depending on what model we are filtering on.

What do you think? Does this make sense for a first pass?